Etabs_Adapter/Helpers/Results.cs

```diff
@@ -511,6 +511,52 @@ private static List<string> CheckAndGetCases(cSapModel model, IList cases)
 
         #endregion
 
+
+        public static List<ModalDynamics> GetModalParticipationMassRatios(cSapModel model, IList cases = null)
+        {
+            List<string> loadcaseIds = new List<string>();
+            List<ModalDynamics> partRatios = new List<ModalDynamics>();
+
+            int resultCount = 0;
+            string[] loadcaseNames = null;
+            string[] stepType = null; double[] stepNum = null;
+            double[] period = null;
+            double[] ux = null; double[] uy = null; double[] uz = null;
+            double[] sumUx = null; double[] sumUy = null; double[] sumUz = null;
+            double[] rx = null; double[] ry = null; double[] rz = null;
+            double[] sumRx = null; double[] sumRy = null; double[] sumRz = null;
+
+            model.Results.ModalParticipatingMassRatios(ref resultCount, ref loadcaseNames, ref stepType, ref stepNum,
+                ref period, ref ux, ref uy, ref uz, ref sumUx, ref sumUy, ref sumUz, ref rx, ref ry, ref rz, ref sumRx, ref sumRy, ref sumRz);
+
+            string previousModalCase = "";
+            int modeNumber = 1;
+
+            for (int i = 0; i < resultCount; i++)
+            {
+                if (loadcaseNames[i] != previousModalCase)
+                    modeNumber = 1;
+
+                ModalDynamics mod = new ModalDynamics()
+                {
+                    ResultCase = loadcaseNames[i],
+                    ModeNumber = modeNumber,
```

Comment:
When pulling modal results, the stepnumber will correspond to the modenumber. THink we should be using that instead of incrementing our own variable. Will slightly simplify the loop and no need to check for "Previous modal case" above.

ref:

http://docs.csiamerica.com/help-files/common-api(from-sap-and-csibridge)/SAP2000_API_Fuctions/Analysis_Results/Results/ModalParticipatingMassRatios.htm

Reply:
Right, should've looked at the API function description, sorry. Thanks for this.